External/Plugins/ASCompletion/Controls/ReferenceList.cs

```diff
@@ -0,0 +1,164 @@
+        {
+            if (!listView.RectangleToScreen(new Rectangle(0, 0, listView.Width, listView.Height)).Contains(Control.MousePosition))
+            {
+                listView.Hide();
+                fadingTimer.Stop();
+            }
+        }
+
+        static void ListView_DoubleClick(object sender, EventArgs e)
+        {
+            var selection = listView.Items[listView.SelectedIndices[0]];
+            var reference = (Reference)selection.Tag;
+
+            listView.Hide();
+            fadingTimer.Stop();
+            PluginBase.MainForm.OpenEditableDocument(reference.File, false);
+            PluginBase.MainForm.CurrentDocument.SciControl.GotoLine(reference.Line);
+        }
+
+        internal static void Show(IEnumerable<Reference> implementors, IEnumerable<Reference> implemented, IEnumerable<Reference> overriders, IEnumerable<Reference> overridden)
+        {
+            listView.Hide();
+            listView.Items.Clear();
+
+            var implementorsGroup = listView.Groups["implementors"];
+            var implementedGroup = listView.Groups["implements"];
+            var overridersGroup = listView.Groups["overriders"];
+            var overriddenGroup = listView.Groups["overrides"];
+
+            AddItems(implementorsGroup, implementors);
+            AddItems(implementedGroup, implemented);
+            AddItems(overridersGroup, overriders);
+            AddItems(overriddenGroup, overridden);
+
+            var p = ((Form)PluginBase.MainForm).PointToClient(Cursor.Position);
+            p.Offset(-2, -2);
+            listView.Location = p;
+
+            listView.Columns[0].AutoResize(ColumnHeaderAutoResizeStyle.ColumnContent);
+            listView.Width = listView.Columns[0].Width + 30;
+
+            listView.BringToFront();
+            listView.Show();
+            listView.Focus();
+            
+            if (listView.Items.Count > 0)
+                listView.Height = Math.Min(listView.Items[listView.Items.Count - 1].Bounds.Bottom + 10, 500);
+
+            fadingTimer.Start();
+            ListView_MouseEnter(null, null);
+        }
+
+        internal static IEnumerable<Reference> ConvertCache(MemberModel member, IEnumerable<ClassModel> list)
+        {
+            return list.Select(m => new Reference
+            {
+                File = m.InFile.FileName,
+                Line = m.Members.Search(member.Name, 0, 0).LineFrom,
+                Type = m.QualifiedName
+            });
+        }
+
+        static void AddItems(ListViewGroup group, IEnumerable<Reference> items)
+        {
+            foreach (var r in items)
+            {
+                var item = new ListViewItem
+                {
+                    Text = r.Type,
+                    Tag = r,
+                    Group = group
+                };
+                listView.Items.Add(item);
+            }
+            
+        }
+    }
+
+    class Reference
```

Comment:
Is not it better to use the structure?

Reply:
I thought about that before.
I went with a class since the structs would become relatively big. Also they will be boxed anyway when they are assigned to the Tag of the item, so performance-wise there is probably not much difference.